### PR TITLE
Form component

### PR DESCRIPTION
**Pull request by RazanTayyem**

### CODE REVIEW

File: student/package.json [hunk 1]
```diff
@@ -4,13 +4,15 @@
   "private": true,
   "dependencies": {
     "eslint-config-react-minimal": "^1.0.0",
+    "formik": "^1.5.1",
     "parcel": "^1.12.3",
```
Comment (mhmdtshref) [inline student/package.json]: We don't need parcel anymore!
Remove it!

File: student/src/components/Button/index.style.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import styled from "styled-components";
+
+export default styled.button`
+background: #38BEFD;
+border-radius: 20px;
+width: 198px;
```
Comment (mhmdtshref) [inline student/src/components/Button/index.style.js]: We shouldn't use static width for elements, use `vw` instead.

File: student/src/components/Button/index.style.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+height: 44px;
```
Comment (mhmdtshref) [inline student/src/components/Button/index.style.js]: Use `vw` here also!
It depends on the width

File: student/src/components/Form/index.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+              {f.label} <br/>
+                {this.generateRadioOptions(f.options)}
+            </StyledLabel>
+          );
+      }
+    });
+    return renderedFields;
+  };
+
+  updateDatesState = (name, value) => {
```
Comment (mhmdtshref) [inline student/src/components/Form/index.js]: Remove All the date-related functions!

**Review by ameentalahmeh — APPROVED** [review]
seen with @mhmdtshref  :+1:

File: student/src/components/Form/index.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    const newFieldsValues = this.state.dateFieldsValues;
+    newFieldsValues[name] = value;
+    this.setState({ dateFieldsValues: newFieldsValues });
+  };
+
+  onFormSubmit = (values, { resetForm }) => {
+    //console.log("Values submitted ::", values);
+    const fullValues = { ...this.props.initialValues, ...values };
+    this.props
+      .action({ ...fullValues, ...this.state.dateFieldsValues })
```
Comment (mhmdtshref) [inline student/src/components/Form/index.js]: Remove date-related options

**Review by mhmdtshref — CHANGES_REQUESTED** [review]

File: student/src/components/Form/index.js (hunk 4 above)
Comment (RazanTayyem) [reply student/src/components/Form/index.js]: :joy_cat:

File: student/src/components/Form/index.js [hunk 6]
```diff
@@ -12,9 +12,9 @@ import {
 } from "./index.style";
 
 class MainForm extends Component {
-  constructor(props) {
-    super(props);
-    this.state = { dateFieldsValues: {} };
+  constructor() {
+    super();
+    this.state={};
```
Comment (mhmdtshref) [inline student/src/components/Form/index.js]: Remove the state if you'll not use it

File: student/src/components/Form/index.js [hunk 7]
```diff
@@ -60,18 +60,11 @@ class MainForm extends Component {
...
-    newFieldsValues[name] = value;
-    this.setState({ dateFieldsValues: newFieldsValues });
-  };
-
   onFormSubmit = (values, { resetForm }) => {
     //console.log("Values submitted ::", values);
     const fullValues = { ...this.props.initialValues, ...values };
     this.props
-      .action({ ...fullValues, ...this.state.dateFieldsValues })
+      .action({ ...fullValues,})
```
Comment (mhmdtshref) [inline student/src/components/Form/index.js]: Here you need to type
`.action(fullValues)`

**Review by mhmdtshref — CHANGES_REQUESTED** [review]
View my comments

**Review by ghassanmas — APPROVED** [review]

**Pull request merged**